src/libsync/syncfilestatustracker.cpp

```diff
@@ -121,10 +122,13 @@ SyncFileStatus SyncFileStatusTracker::fileStatus(const QString& relativePath)
 void SyncFileStatusTracker::slotPathTouched(const QString& fileName)
 {
     QString folderPath = _syncEngine->localPath();
-    Q_ASSERT(fileName.startsWith(folderPath));
 
-    QString localPath = fileName.mid(folderPath.size());
-    _dirtyPaths.insert(localPath);
+    if (fileName.startsWith(folderPath)) {
+        QString localPath = fileName.mid(folderPath.size());
+        _dirtyPaths.insert(localPath);
+    } else {
```

Comment:
I would prefer not to add an if, this code path is not valid and there is no need to prevent the above code to run for an impossible situation.